### ISSUE TITLE
Booleans writing to mongodb as strings

### ISSUE THREAD

**Issue by jeffreykrodgers** [meteor/meteor#8528]
I updated my project to Meteor 1.4.3.2 and ever since then, whenever I use a method to update a value in my database, it writes it as a string instead of a boolean. I am checking the typeof all the way up to the point that the db is updated and it always returns boolean, but when I check the db itself the value is going in as a string type. 

Here's an example:
```
Meteor.methods({
        "df_apply_dark_mode": (user, mode) => {
            df_in_check(String, [user]);
            df_in_check(Boolean, [mode]);

            console.log(typeof mode); // this returns boolean

            Meteor.users.update(user, {
                $set: {
                    'profile.dark_mode': mode,
                },
            });
        },
});
```

When I then check my database directly, profile.dark_mode is a string. Am I doing something wrong or is this a bug?

Here are my packages:
```
accounts-facebook            1.1.1  Login service for Facebook accounts
accounts-password            1.3.4  Password support for accounts
accounts-ui                  1.1.9  Simple templates to add login widgets to an app
aldeed:autoform              5.8.1* Easily create forms with automatic insert and update, and automatic reactive validation.
aldeed:collection2           2.10.0  Automatic validation of insert and update operations on the client and server.
audit-argument-checks        1.0.7  Try to detect inadequate input sanitization
blaze-html-templates         1.1.2  Compile HTML templates into reactive UI with Meteor Blaze
browser-policy               1.1.0  Configure security policies enforced by the browser
check                        1.2.5  Check whether a value matches a pattern
dburles:collection-helpers   1.1.0  Transform your collections with helpers that you define
ecmascript                   0.6.3  Compiler plugin that supports ES2015+ in all .js files
ejson                        1.0.13  Extended and Extensible JSON library
es5-shim                     4.6.15  Shims and polyfills to improve ECMAScript 5 support
facebook-config-ui           1.0.0  Blaze configuration templates for Facebook OAuth.
fastclick                    1.0.13  Faster touch events on mobile
johnantoni:meteor-normalize  0.0.1  normalize.css ~ packaged for Meteor
jquery                       1.11.10  Manipulate the DOM using CSS selectors
jshint                       1.1.7  Lint all your JavaScript files with JSHint.
juliancwirko:postcss         1.2.0  Minifier for Meteor with PostCSS processing - use Autoprefixer and others with ease
kadira:blaze-layout          2.3.0  Layout Manager for Blaze (works well with FlowRouter)
kadira:flow-router           2.12.1  Carefully Designed Client Side Router for Meteor
less                         2.7.9  Leaner CSS language
logging                      1.1.17  Logging facility.
meteor-base                  1.0.4  Packages that every Meteor app needs
meteorhacks:kadira           2.30.3  Performance Monitoring for Meteor
mobile-experience            1.0.4  Packages for a great mobile user experience
```

**Comment by hwillson** [comment]
Hi @jeffreykrodgers - sorry to hear you're having trouble! Are you able to create a small, runnable reproduction (with as few dependencies as possible) demonstrating this issue (see [Reporting a bug in Meteor](https://github.com/meteor/meteor/blob/devel/Contributing.md#reporting-bug))? This should work properly, so narrowing this down in a small reproduction will greatly help with troubleshooting. Thanks!

**Comment by jeffreykrodgers** [comment]
Aaah it was my mistake. Ugh. Found it while I was making that small, runnable reproduction.

The issue was coming from the collection2 package. I had a schema attached to the users that had 'dark_mode' listed as a string, not a boolean. Not sure why it worked all the way up to this point, but once I changed the type from string to boolean, it started working.

In case anyone else makes the same mistake, there it is.

Thanks!
